### DISCUSSION TITLE
[4.0] Create Namespace extension earlier then onExtensionAfterInstall

### DISCUSSION THREAD

**Discussion by 810** [joomla/joomla-cms#47164] [2. Cold Features]
### Steps to reproduce the issue

I have a package with admin/site/library/plugins/media installers.

I want to use a namespace on the pkg script on the postflight.

````
use Kunena\Forum\Libraries\Install\KunenaModelInstall;
````

On the postflight:
````
$installer = new KunenaModelInstall;
$installer->createMenu();
````

### Expected result

That the namespace is already registered at autoload_psr4 file.

### Actual result

Class 'Kunena\Forum\Libraries\Install\KunenaModelInstall' not found

### System information (as much as possible)

Nightly build J4

Kunena 6.0 nightly build:  www.kunena.org/download on the bottom is the nightly build.

**Comment by xillibit** [comment]
Someone has an idea on this ?

**Comment by xillibit** [comment]
Tested on beta4-dev build, still lost the id of component during installation of component : 

![image](https://user-images.githubusercontent.com/876395/91634390-5d012380-e9f0-11ea-81c4-35d158b4ec75.png)


So nobody has an idea of what going on ?

**Comment by xillibit** [comment]
We have found a workaround for it : https://github.com/Kunena/Kunena-Forum/commit/85c63867ac86ee49270bac1c068d652a7aa4d57c

**Comment by zero-24** [comment]
I guess the best place would be somewhere here: https://github.com/joomla/joomla-cms/blob/4.0-dev/libraries/src/Installer/InstallerAdapter.php#L1005 right? So the new names are loaded just before the script.php is executed right?

**Comment by xillibit** [comment]
I'am agree with you

**Comment by zero-24** [comment]
Can you do a PR? So we can finally fix this in the core and you can get rid of your workaround?
